### PR TITLE
Docs: Improve KTextbox component documentation with examples

### LINKED ISSUE #596: Add examples to KTextbox documentation
## Overview

@GarvitSinghal47 pointed out that some of our components' docs could be improved with some examples, starting with KTextbox.

Not all variations need to be applied, however, a handful of the props' examples would be a great improvement.

## Acceptance criteria

The following examples can be made initially and further updates to component docs can be discussed afterward:

- [ ] `invalid` and `invalidText` -- this will need to have a note to the user regarding what is and isn't valid so that they can test it out
- [ ] `disabled` 
- [ ] `number` `min` `max` 
- [ ] `textarea`
- [ ] `label` 

---

I think that going beyond 5 examples might ultimately make the page less usable/readable overall but once these are in place we can discuss further.

### PR DESCRIPTION
**Pull request by GarvitSinghal47**
Fixes #596 

## Description

This pull request enhances the documentation of our component by adding initial examples for selected props. Not all variations are included, but we've added examples for 'invalid' and 'invalidText' with notes for user guidance, 'disabled', 'number' with 'min' and 'max' attributes, 'textarea', and 'label'. This improvement aims to make the component documentation more comprehensive and user-friendly. Further updates and discussions on additional examples can be made after these initial changes.

## Changelog

- [#PR no]
  - **Description:** Add initial examples for selected props in component documentation
  - **Products impact:** none
  - **Addresses:** none
  - **Components:** Documentation
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** These changes enhance the documentation by providing initial examples for selected props. Further updates and discussions can be conducted after these initial changes.

## Steps to test

1. Navigate to the component documentation.
2. Look for examples provided for the mentioned props: 'invalid', 'invalidText', 'disabled', 'number', 'textarea', and 'label'.
3. Verify that the examples are correctly displayed and include any necessary guidance notes.

### At a high level, how did you implement this?

These changes involve updating the documentation markdown files to include examples for the selected props.

### Does this introduce any tech-debt items?

No, these changes do not introduce any tech debt items.

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical and brittle code paths are covered by unit tests
- [x] The change is described in the changelog section above

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] Are there tests for this change?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?

## After review

- [ ] The changelog item has been pasted to the `CHANGELOG.md`

### CODE REVIEW

**Comment by MisRob** [comment]
Thank you, @GarvitSinghal47. Could you retarget to `release-v4` branch?

File: CHANGELOG.md
Comment (MisRob) [inline CHANGELOG.md]: Thank you, @GarvitSinghal47! I will leave the review to @nucleogenesis, just chiming in to ask if you could remove this change (probably caused by rebasing branches?)

File: docs/pages/ktextbox.vue
Comment (nucleogenesis) [inline docs/pages/ktextbox.vue]: Here it might be good to also note that it cannot be focused by way of keyboard navigation, so perhaps:

"This text box is disabled. It cannot be edited or focused, so it will be skipped during keyboard navigation."

Thoughts @radinamatic @MisRob ?

File: docs/pages/ktextbox.vue
Comment (nucleogenesis) [inline docs/pages/ktextbox.vue:47]: _Non-blocking_

I wonder if KTextbox should be setting its own state to an error based on the given min & max values. Currently the user has to implement the checks themselves but seems like it would make sense to provide a default `invalid` checking function. 

Also despite the min & max values being set here, I don't see any changes or feedback when I put an invalid number -- so if that's intended, perhaps we should note that in the example? ie, "KTextbox does not automatically validate the min and max values, however, it is recommended to use them for accessibility purposes while using the `invalid` and `invalidText` props to handle validation as needed"

**Review by nucleogenesis — CHANGES_REQUESTED** [review]
Thank you @GarvitSinghal47 this is great. I left a couple comments and only one is blocking re: adding a note about the focus-ability of disabled inputs.

Also looks like the linter found something as well.

File: docs/pages/ktextbox.vue
Comment (nucleogenesis) [inline docs/pages/ktextbox.vue]: I think that since the `label` prop is `required` that we should remove this one since it results in a console error.

File: docs/pages/ktextbox.vue
Comment (GarvitSinghal47) [reply docs/pages/ktextbox.vue:47]: @nucleogenesis Absolutely, incorporating automatic error state handling for the KTextbox component based on the provided min and max values does sound like a beneficial enhancement. By implementing a default invalid checking function within KTextbox, users can enjoy a smoother validation experience without having to manually handle it each time.

For the second part, should we simply add a comment to the example code explaining this behavior, or do you think it would be more helpful to include a function that displays an error message when an invalid number is entered? Let me know your thoughts on how best to proceed.

File: docs/pages/ktextbox.vue
Comment (MisRob) [reply docs/pages/ktextbox.vue]: Makes sense to me

**Comment by MisRob** [comment]
Thanks both @GarvitSinghal47 and @nucleogenesis, looking great and good suggestions.

This is nothing blocking, just a suggestion for possible improvement. I think it's always helpful to show how code for examples look like. Demonstration of this can be seen in some newer documentation pages, e.g. that of for `KImg`. KDS system has a documentation component for this purpose ready - you can see how `<DocsShowCode language=...>` is being used

File: docs/pages/ktextbox.vue
Comment (MisRob) [reply docs/pages/ktextbox.vue:47]: @GarvitSinghal47

I think for now we could go with @nucleogenesis's suggestion to include a comment in the documentation:

> "KTextbox does not automatically validate the min and max values, however, it is recommended to use them for accessibility purposes while using the invalid and invalidText props to handle validation as needed"

and open an issue for the error state handling.

**Review by nucleogenesis — APPROVED** [review]
The latest changes LGTM! Love seeing the code examples there :) thanks Garvit -- I'll leave it to @MisRob to decide when to merge from here.

File: docs/pages/ktextbox.vue
Comment (MisRob) [inline docs/pages/ktextbox.vue]: Could you please update to

```suggestion
    <DocsPageSection title="Usage">
```

This uses language from other documentation pages. Whenever possible we'd want to re-use section titles, it will help with overall consistency.

**Comment by MisRob** [comment]
Thanks both, it's looking great overall. Just one last thing before we merge.

**Review by MisRob — APPROVED** [review]
Thanks @GarvitSinghal47 and @nucleogenesis, this is a valuable improvement of the documentation

**Pull request merged**
